src/player.cpp

```diff
@@ -4045,6 +4048,19 @@ void player::mod_pain(int npain) {
         if( has_trait( trait_NOPAIN ) ) {
             return;
         }
+        if( has_trait( trait_MOREPAIN ) || has_trait ( trait_MOREPAIN2 ) || has_trait ( trait_MOREPAIN3 ) ){
```

Comment:
You can get rid of this check and replace the `else` leg of the nested condition with `else if ( has_trait ( trait_MOREPAIN3 ) )`.

Reply:
And instead of calculating `npain` and comparing to `oldpain`, just calculate the modifier

```
int pain_mod = 0;
if (has_trait ( trait_MOREPAIN ) ) {
  pain_mod = roll_remainder( npain * 0.25f );
} else
  ....
}

npain += min(1, pain_mod);
````

Reply:
Updated this, thanks for the comments folks. Btw, think you meant max, not min.